### ISSUE TITLE
Why need to create new login and password to login as watch only mode?

### ISSUE THREAD

**Issue by bitcoinuser** [greenaddress/GreenBits#159]
Hi, I see greenbits added watch only login mode, where is necessary to create a new login and password.

I think watch only login mode shouldn't request to add login and password. I would like to start greenbits as watch only login mode without be necessary to insert passwords or login. Is it possible?

**Comment by greenaddress** [comment]
Can you describe in more details what you would expect from something like this?

At the moment the watch only can be setup to remember the username but not the password, do you want to avoid to have to enter the password (i.e. save that with the username) or you don't want to have to create a user/pwd at all in first place (let alone use them during login) ?

Thanks

**Comment by bitcoinuser** [comment]
Ok,

1- My greenbits wallet is 2of3
2- I can't login on greenbits without internet connection because greenbits use the PIN to send the password used to decrypt the mnemnonic passphrase stored on user device. 
Question: Why this mechanism can' work locally on each user device?

The way (more and less) how I would like to see greenbits work:

1- I create a 2of3 wallet on Greenbits.
2-  I insert the mnemnonic passphrase and create a PIN.
3-  Greenbits will store on my device the mnemnonic passphrase encrypted. To decrypt it, the user will need to insert the PIN. If the PIN is provided wrong for 3 times, the encrypted mnemnonic passphrase stored on user device will be deleted.
4- Greenbits will too store (on user device), the xpub of the private key that only greenbits have access.

In this way users could login in watch only mode without internet connection, because the user mnemnonic passphrase key and the greenbits xpub key will be stored on user device. So I could generate public address without internet connection. So If the user wants to spend the bitcoins, he will need to provide the PIN and probably a 2FA. 

What is wrong with this model?

**Comment by greenaddress** [comment]
I will reply to your questions with more details later but for now I can say that one of the issue (at least for now) is that the service needs to know which addresses you generated so that it can monitor them, send you notification (email/web socket) etc

We'll think more about it, thanks

**Comment by bitcoinuser** [comment]
Hum, If you have one of the keys of a 2of3 wallet, can't you monitor the address generated? I didn't know about this.

**Comment by bitcoinuser** [comment]
But maybe Greenbits app can send the xpub of the private key created locally on users devices. So greenbits will have access to two xpub keys. I imagine that with two xpub keys, Greenbits can monitor the address generated, no?

**Comment by greenaddress** [comment]
You can in theory, you just try to generate the next address and see if anything comes but how does the service know when to stop? It would have to try to generate all addresses and they are infinite. We could stop after a number of addresses if we don't find anything but I never liked that solution since you just don't know for sure.

Does it make sense?

**Comment by greenaddress** [comment]
It's not about GreenBits, GreenBits has all it needs to create addresses, it's about the service knowing what to notify you on.

**Comment by bitcoinuser** [comment]
Ok, so:

Why not let the users open the app in offline mode? The users will not be able to generate new addresses or send bitcoins, but could see the last generated address and see the last transactions

Make sense?

Thanks.

**Comment by greenaddress** [comment]
Yes that's possible, but that data needs to be kept somewhere (on the device?) and should it be encrypted? if so, with what password? If not encrypted, isn't that bad and a weakness, privacy etc?

**Comment by bitcoinuser** [comment]
Encrypted with the same PIN used now. Could have an option on config asking if the user wants to store the data offline.

**Comment by greenaddress** [comment]
So the PIN is ok with the server because the server can delete the associated password after 3 wrong attempts, but if we use the PIN to encrypt the data it is then trivial to bruteforce locally and even if the app deletes the data you can backup the data before that happens and basically try infinite times.

It wouldn't take very long at all to brute force the PIN - it's almost better to not have it to not provide a false sense of security.

Thanks!

**Comment by bitcoinuser** [comment]
> If not encrypted, isn't that bad and a weakness, privacy etc?

But I don't think it's bad. The user will not be able to spend the funds. The data will be stored only on my device. What could go wrong? If someone has access to my Android, many things will be privacy compromised. The important thing is not be able to spend the funds.
And as I said, this option to could optional and per device.
